src/query.rs

```diff
@@ -80,13 +124,30 @@ where
         }
     }
 
+    /// Narrow down a query to entities that have a certain component,
+    /// without obtaining that component.
```

Comment:
Maybe we should strengthen this as the component is not only not fetch, but also not borrowed?